js/app/models/veventModel.js

```diff
@@ -0,0 +1,289 @@
+			ICAL.TimezoneService.register(timezone.tzid, timezone);
+		});
+
+		if (!uri) {
+			const vevent = context.comp.getFirstSubcomponent('vevent');
+			context.uri = vevent.getFirstPropertyValue('uid');
+		}
+
+		/**
+		 * get DTEND from vevent
+		 * @param {ICAL.Component} vevent
+		 * @returns {ICAL.Time}
+		 */
+		context.calculateDTEnd = function(vevent) {
+			if (vevent.hasProperty('dtend')) {
+				return vevent.getFirstPropertyValue('dtend');
+			} else if (vevent.hasProperty('duration')) {
+				const dtstart = vevent.getFirstPropertyValue('dtstart').clone();
+				dtstart.addDuration(vevent.getFirstPropertyValue('duration'));
+
+				return dtstart;
+			} else {
+				return vevent.getFirstPropertyValue('dtstart').clone();
+			}
+		};
+
+		/**
+		 * convert a dt's timezone if necessary
+		 * @param {ICAL.Time} dt
+		 * @param {ICAL.Component} timezone
+		 * @returns {ICAL.Time}
+		 */
+		context.convertTz = function(dt, timezone) {
+			if (context.needsTzConversion(dt) && timezone) {
+				dt = dt.convertToZone(timezone);
+			}
+
+			return dt;
+		};
+
+		/**
+		 * check if we need to convert the timezone of either dtstart or dtend
+		 * @param {ICAL.Time} dt
+		 * @returns {boolean}
+		 */
+		context.needsTzConversion = function(dt) {
+			return (dt.icaltype !== 'date' &&
+				dt.zone !== ICAL.Timezone.utcTimezone &&
+				dt.zone !== ICAL.Timezone.localTimezone);
+		};
+
+		Object.defineProperties(iface, {
+			calendar: {
+				get: function() {
+					return context.calendar;
+				},
+				set: function(calendar) {
+					context.calendar = calendar;
+				}
+			},
+			comp: {
+				get: function() {
+					return context.comp;
+				}
+			},
+			data: {
+				get: function() {
+					return context.comp.toString();
+				}
+			},
+			etag: {
+				get: function() {
+					return context.etag;
+				},
+				set: function(etag) {
+					context.etag = etag;
+				}
+			},
+			uri: {
+				get: function() {
+					return context.uri;
+				}
+			}
+		});
+
+		/**
+		 * get fullcalendar event in a defined time-range
+		 * @param {moment} start
+		 * @param {moment} end
+		 * @param {Timezone} timezone
+		 * @returns {Array}
+		 */
+		iface.getFcEvent = function(start, end, timezone) {
+			const iCalStart = ICAL.Time.fromJSDate(start.toDate());
+			const iCalEnd = ICAL.Time.fromJSDate(end.toDate());
+			const fcEvents = [];
+
+			const vevents = context.comp.getAllSubcomponents('vevent');
+			vevents.forEach(function(vevent) {
+				const iCalEvent = new ICAL.Event(vevent);
+
+				if (!vevent.hasProperty('dtstart')) {
+					return;
+				}
+
+				const rawDtstart = vevent.getFirstPropertyValue('dtstart');
+				const rawDtend = context.calculateDTEnd(vevent);
+
+				if (iCalEvent.isRecurring()) {
+					const duration = rawDtend.subtractDate(rawDtstart);
+					const iterator = new ICAL.RecurExpansion({
+						component: vevent,
+						dtstart: rawDtstart
+					});
+
+					let next;
+					while ((next = iterator.next())) {
```

Comment:
I'd like to keep this actually, because we need to evaluate the result of the assignment.
I'm not sure if it works in all browsers when using `while(next=iterator.next()){...}`